benchmark/experiments/comps/Orth_speed.cc

```diff
@@ -0,0 +1,253 @@
+        std::vector<T> A_cpy_2(size, 0.0);
+        std::vector<T> A_cpy_3(size, 0.0);
+	    
+        T* A_dat = A.data();
+        T* A_cpy_dat = A_cpy.data();
+        T* A_cpy_2_dat = A_cpy_2.data();
+        T* A_cpy_3_dat = A_cpy_3.data();
+
+        // Random Gaussian test matrix
+        RandBLAS::dense_op::gen_rmat_norm<T>(m, n, A_dat, seed);
+        // Make a copy
+        std::copy(A_dat, A_dat + size, A_cpy_dat);
+        std::copy(A_dat, A_dat + size, A_cpy_2_dat);
+        std::copy(A_dat, A_dat + size, A_cpy_3_dat);
+
+        // Stabilization Constructor
+        RandLAPACK::comps::orth::Stab<T> Stab_PLU(1);
+        RandLAPACK::comps::orth::Orth<T> Orth_CholQR(0);
+        RandLAPACK::comps::orth::Orth<T> Orth_HQR(1);
+        RandLAPACK::comps::orth::Orth<T> Orth_GEQR(2);
+
+        // PIV LU
+        // Stores L, U into Omega
+        auto start_lu = high_resolution_clock::now();
+        Stab_PLU.call(m, n, A_cpy);
+        auto stop_lu = high_resolution_clock::now();
+        long dur_lu = duration_cast<microseconds>(stop_lu - start_lu).count();
+
+        // HQR
+        auto start_qr = high_resolution_clock::now();
+        Orth_HQR.call(m, n, A_cpy_2);
+        auto stop_qr = high_resolution_clock::now();
+        long dur_qr = duration_cast<microseconds>(stop_qr - start_qr).count();
+
+        // CHOL QR
+        // Orthonormalize A
+        auto start_chol = high_resolution_clock::now();
+        Orth_CholQR.call(m, n, A);
+        auto stop_chol = high_resolution_clock::now();
+        long dur_chol = duration_cast<microseconds>(stop_chol - start_chol).count();
+
+        // GEQR
+        auto start_geqr = high_resolution_clock::now();
+        Orth_GEQR.call(m, n, A);
+        auto stop_geqr = high_resolution_clock::now();
+        long dur_geqr = duration_cast<microseconds>(stop_geqr - start_geqr).count();
+
+
+        return std::make_tuple(dur_chol, dur_lu, dur_qr, dur_geqr);
+    }
+
+    template <typename T>
+    static void 
+    test_speed(int r_pow, int r_pow_max, int c_pow, int c_pow_max, int runs)
+    {
+        int64_t rows = 0;
+        int64_t cols = 0;
+
+        T chol_avg = 0;
+        T lu_avg = 0;
+        T qr_avg = 0;
+        T geqr_avg = 0;
+
+        for(; r_pow <= r_pow_max; ++r_pow)
+        {
+            rows = std::pow(2, r_pow);
+            int c_buf = c_pow;
+
+            for (; c_buf <= c_pow_max; ++c_buf)
+            {
+                cols = std::pow(2, c_buf);
+
+                std::tuple<long, long, long, long> res;
+                long t_chol = 0;
+                long t_lu   = 0;
+                long t_qr   = 0;
+                long t_geqr = 0;
+
+                long curr_t_chol = 0;
+                long curr_t_lu   = 0;
+                long curr_t_qr   = 0;
+                long curr_t_geqr = 0;
+
+                std::ofstream file("../../build/test_plots/test_speed/raw_data/test_" + std::to_string(rows) + "_" + std::to_string(cols) + ".dat");
```

Comment:
If we use relative file paths then we'll want to make sure C++ doesn't fail silently. That is, we'll want to raise an error if a file doesn't exist. But we can probably come back to this later.

Reply:
Note in TODO taken